main/src/com/google/refine/expr/functions/EpochToDate.java

```diff
@@ -0,0 +1,66 @@
+
+package com.google.refine.expr.functions;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.Instant;
+import java.util.Properties;
+import com.google.refine.expr.EvalError;
+import com.google.refine.grel.ControlFunctionRegistry;
+import com.google.refine.grel.Function;
+
+public class EpochToDate implements Function {
+
+    @Override
+    public Object call(Properties bindings, Object[] args) {
+        if (args.length >= 1 && args[0] instanceof Number) {
+            long epoch = (long) args[0];
+            ZoneId zoneId = ZoneId.of("UTC");
+            Instant instant = null;
+            OffsetDateTime date = null;
+            if (args.length == 1) {
+                instant = Instant.ofEpochSecond(epoch);
+                date = OffsetDateTime.ofInstant(instant, zoneId);
+                return date;
+            } else if (args.length == 2 && args[1] instanceof String) {
+                Object o2 = args[1];
+                String type = ((String) o2).toLowerCase();
+                if (type.equals("second")) {
+                    instant = Instant.ofEpochSecond(epoch);
+                    date = OffsetDateTime.ofInstant(instant, zoneId);
+                    return date;
+                } else if (type.equals("millisecond")) {
+                    instant = Instant.ofEpochSecond(epoch / 1000);
+                    date = OffsetDateTime.ofInstant(instant, zoneId);
+                    return date;
+                } else if (type.equals("microsecond")) {
+                    instant = Instant.ofEpochSecond(epoch / 1000000);
+                    date = OffsetDateTime.ofInstant(instant, zoneId);
+                    return date;
+                }
+            }
+        }
+        return new EvalError(ControlFunctionRegistry.getFunctionName(this) + " expect one argument, " +
+                "which is a number of epoch time\n" +
+                "expect two argument, the first is a epoch time(second, millisecond, microsecond), the second " +
+                "is the type");
+    }
+
+    @Override
+    public String getDescription() {
```

Comment:
Again suggest that this could be slightly more concise. I suggest reviewing https://github.com/OpenRefine/OpenRefine/blob/235b5957e22c73328e9f84d6fac44a9cb92318f5/main/src/com/google/refine/expr/functions/strings/Phonetic.java#L104 for an example of a description.

e.g.
"Returns a number converted to a date based on Unix Epoch Time. The number can be Unix Epoch Time in one of the following supported units: second, millisecond, microsecond. Defaults to 'decode'."